pennylane/qnode.py

```diff
@@ -250,20 +257,22 @@ def construct(self, args, **kwargs):
             args (tuple): Represent the free parameters passed to the circuit.
                 Here we are not concerned with their values, but with their structure.
                 Each free param is replaced with a :class:`~.variable.Variable` instance.
-
-        .. note::
-
-            Additional keyword arguments may be passed to the quantum circuit function, however PennyLane
-            does not support differentiating with respect to keyword arguments. Instead,
-            keyword arguments are useful for providing data or 'placeholders' to the quantum circuit function.
+            kwargs (dict): Additional keyword arguments may be passed to the quantum circuit function,
+                however PennyLane does not support differentiating with respect to keyword arguments.
+                Instead, keyword arguments are useful for providing data or 'placeholders'
+                to the quantum circuit function.
         """
-        # pylint: disable=too-many-branches
+        # pylint: disable=too-many-branches,too-many-statements
         self.queue = []
         self.ev = []  # temporary queue for EVs
 
+        if kwargs is None:
+            kwargs = {}
+
         # flatten the args, replace each with a Variable instance with a unique index
         temp = [Variable(idx) for idx, val in enumerate(_flatten(args))]
         self.num_variables = len(temp)
+        self.args_shape = args
```

Comment:
this is kind of a weird usage :thinking:

Reply:
I guess it's meant to be the reference shape, including nested structure?

Reply:
Yep, I just need the shape, but not the actual arguments! There might be a better way of doing this, but we can't assume NumPy here which makes it tricky. Perhaps this is still the cleanest solution?

Reply:
maybe just use a better name? `args_ref`?

Reply:
internal flatten function uses the name `model`